apps/indexer/src/api/repositories/last-update.repository.ts

```diff
@@ -0,0 +1,45 @@
+import { db } from "ponder:api";
+import { inArray } from "ponder";
+import { daoMetricsDayBucket } from "ponder:schema";
+
+import { ChartType } from "../mappers/last-update";
+import { MetricTypesEnum } from "@/lib/constants";
+
+export class LastUpdateRepository {
+  async getLastUpdate(chart: ChartType) {
+    let metricsToCheck: MetricTypesEnum[] = [];
+
+    // Determine which metrics to check based on chart type
+    switch (chart) {
+      case ChartType.CostComparison:
+        metricsToCheck = [MetricTypesEnum.DELEGATED_SUPPLY];
+        break;
+      case ChartType.AttackProfitability:
+        metricsToCheck = [
+          MetricTypesEnum.DELEGATED_SUPPLY,
+          MetricTypesEnum.TREASURY,
+        ];
+        break;
+      case ChartType.TokenDistribution:
+        metricsToCheck = [
+          MetricTypesEnum.DELEGATED_SUPPLY,
+          MetricTypesEnum.CEX_SUPPLY,
+          MetricTypesEnum.DEX_SUPPLY,
+          MetricTypesEnum.LENDING_SUPPLY,
+        ];
+        break;
+      default:
+        break;
+    }
+    // Find the record with the greatest timestamp for the specified metrics
+    const lastUpdate = await db.query.daoMetricsDayBucket.findFirst({
+      columns: {
+        date: true,
+      },
+      where: inArray(daoMetricsDayBucket.metricType, metricsToCheck),
+      orderBy: (fields, { desc }) => [desc(fields.date)],
+    });
```

Comment:
This won't work its purpose since we reset the time of each daily metric to midnight

<img width="501" height="385" alt="Image" src="https://github.com/user-attachments/assets/d27734e8-2581-4a5c-bee3-15bd7fe42673" />